### PR TITLE
Improved the SQL queries to allow filtering applications with multiple filter attributes

### PR DESCRIPTION
**Pull request by ravinduramesh**
### Proposed changes in this pull request
Improved the SQL queries to allow filtering applications by using any validated and preprocessed where clause that can contain multiple filter attributes sent by the API.

Resolves [#10990](https://github.com/wso2-enterprise/asgardeo-product/issues/10990), [#11371](https://github.com/wso2-enterprise/asgardeo-product/issues/11371), [#13933](https://github.com/wso2/product-is/issues/13933)

### When should this PR be merged
Merge with [related pull request in identity-api-server repo](https://github.com/wso2/identity-api-server/pull/377) 

#### Tests
Should test the filtering applications feature in application listing page with following databases.
- [x] 1. MySQL, H2, MariaDB
- [x] 2. PostgreSQL
- [x] 3. Oracle
- [x] 4. MS SQL Server
- [x] 5. DB2
- [x] 6. InfoMix

### CODE REVIEW

**Review by kayathiri4 — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java [hunk 1]
```diff
@@ -49,82 +50,150 @@ public int getApplicationId() {
     /**
      * Set application id.
      *
-     * @param applicationId application id
+     * @param applicationId application id.
```
Comment (Thisara-Welmilla) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:53]: >      * @param applicationId		Application id.

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java [hunk 2]
```diff
@@ -49,82 +50,150 @@ public int getApplicationId() {
...
+
+        return inboundKey;
+    }
+
+    /**
+     * Set inbound auth key.
+     *
+     * @param inboundKey    inbound auth key.
+     */
+    public  void setInboundKey(String inboundKey) {
```
Comment (Thisara-Welmilla) [inline components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:195]: Is there two space after `public` :)

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java [hunk 3]
```diff
@@ -3357,6 +3386,85 @@ private String resolveSQLFilter(String filter) {
...
+
+                    filterData.setFilterString(formattedLeftNodeString + " " +
+                            operationNode.getOperation() + " " + formattedRightNodeString);
+                    filterData.addFilterValue(generateFilterValueForBackend(expressionLeftNode.getOperation(),
+                            expressionLeftNode.getValue()));
+                    filterData.addFilterValue(generateFilterValueForBackend(expressionRightNode.getOperation(),
+                            expressionRightNode.getValue()));
+                }
+            } catch (IOException | IdentityException e) {
+                log.error("Error occurred while converting filter for backend.", e);
```
Comment (AnuradhaSK) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:3428]: why doesn't this error is thrown to upper levels? What's the API response if the request go into this catch?

Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java:3428]: Resolved in https://github.com/wso2/carbon-identity-framework/pull/4432

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java (hunk 2 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:195]: Resolved in https://github.com/wso2/carbon-identity-framework/pull/4432

File: components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java (hunk 1 above)
Comment (ravinduramesh) [reply components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ApplicationBasicInfo.java:53]: Resolved in https://github.com/wso2/carbon-identity-framework/pull/4432

**Pull request merged**